crates/task-impls/src/transactions.rs

```diff
@@ -165,20 +165,28 @@ impl<
                         .number_of_empty_blocks_proposed
                         .add(1);
 
+                    let validated_state = self.consensus.read().await.decided_state();
+
                     // Calculate the builder fee for the empty block
                     let Some(builder_fee) = null_block::builder_fee(
                         self.membership.total_nodes(),
+                        validated_state.as_ref(),
                         self.instance_state.as_ref(),
-                    ) else {
+                    )
```

Comment:
done in 2d78d1e460166d04d8af35cd2671757e032af854

Reply:
where do I keep the handle of this task?

Reply:
You could do something like the consensus task does:
```rust
    pub spawned_tasks: BTreeMap<TYPES::Time, Vec<JoinHandle<()>>>,
```

So we can keep a map around and key on the view or whatever relevant key we need, then we can cancel the task from there. The key here cancels old tasks if we propose for a newer view, so in this case we can probably cancel when we get a new view change? We could cancel with something reminiscent of the following: https://github.com/EspressoSystems/HotShot/blob/main/crates/task-impls/src/consensus/mod.rs#L151 but we might be able to just cancel all tasks older than `view`. Thoughts @bfish713?

Reply:
I think this shouldn't be canceled on a view change since we should be able to complete catch up even if there is a view change to keep the updated chain config in ValidatedState.

Reply:
could be canceled after a certain time maybe

Reply:
I definitely think we shouldn't drop the handle, though I don't think `TaskRegistry` is the right place to put this (it will likely be impossible to put it there very soon). It should definitely be stored in `TransactionTaskState`  somehow, probably either in a `Vec<JoinHandle<()>>` or a `BTreeMap` like Jarred suggested.

I guess the question is: what is the expected behavior if consensus is on view (say) 100 and a calculation for view 50 is still going? do we just keep going indefinitely and risk accumulating these spawned subtasks? or is there a point where we can just give up? (I don't know enough to know what the answer is)

Reply:
reverted this change as builder_fee function is synchronous